test/spec/duplexSpec.js

```diff
@@ -1426,8 +1426,47 @@ describe('duplex', function() {
       }, 20);
     });
   });
-
   describe('compare', function() {
+    it('Replacing a root array with an object should be handled well', function() {
+
+      const obj = {};
+      var patches = jsonpatch.compare(['jack'], obj);
+      expect(patches).toEqual([
+        {
+          op: 'replace',
+          path: '',
+          value: obj
+        }
+      ]);
+
+    });
+    it('Replacing a deep array with an object should be handled well', function() {
```

Comment:
It's rather an array nested in an object, than deep array, as it's only shallow - one level array of primitives.